src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -3683,7 +3683,7 @@ public object VisitPipeline(PipelineAst pipelineAst)
             var temps = new List<ParameterExpression>();
             var exprs = new List<Expression>();
 
-            if (!(pipelineAst.Parent is AssignmentStatementAst || pipelineAst.Parent is ParenExpressionAst))
+            if (pipelineAst.Parent is AssignmentStatementAst || pipelineAst.Parent is not ParenExpressionAst)
```

Comment:
This logic should match the original:

```suggestion
            if (pipelineAst.Parent is not AssignmentStatementAst && pipelineAst.Parent is not ParenExpressionAst)
```